webapi/webapi-sensors-w3c-tests/sensors/resources/generic-sensor-tests.js

```diff
@@ -0,0 +1,149 @@
+function runGenericSensorTests(sensorType, readingType, verifyReading, arrReading) {
+  async_test(t => {
+    let sensor = new sensorType();
+    sensor.start();
+
+    sensor.onchange = t.step_func_done(() => {
+      assert_true(verifyReading(sensor.reading));
+      assert_equals(sensor.state, "activated");
+      sensor.stop();
+    });
+
+    sensor.onerror = t.step_func_done(event => {
+      assert_unreached(event.error.name + ":" + event.error.message);
+    });
+  }, "event change fired");
+
+  async_test(t => {
+    let sensor = new sensorType();
+    sensor.start();
+    sensor.onactivate = t.step_func_done(() => {
+      let cachedReading = sensor.reading;
+      let cached1 = arrReading(cachedReading);
+      sensor.stop();
+      let cached2 = arrReading(cachedReading);
+      for (i=0;i<cached2.length;i++) {
+        assert_equals(cached1[i], cached2[i]);
+      }
+    });
+    sensor.onerror = t.step_func_done(event => {
+      assert_unreached(event.error.name + ":" + event.error.message);
+    });
+  }, "Test that sensor reading must be immutable.");
+
+  async_test(t => {
+    sensor1 = new sensorType();
+    sensor2 = new sensorType();
+    sensor1.start();
+    sensor2.start();
+    sensor1.onactivate = t.step_func_done(() => {
+      cachedReading1 = sensor1.reading;
+      cachedReading2 = sensor2.reading;
+      //both sensors share the same reading instance
+      assert_equals(cachedReading1, cachedReading2);
+      //after first sensor stops its reading is null, second sensor remains
+      sensor1.stop();
+      assert_equals(sensor1.reading, null);
+      assert_true(sensor2.reading instanceof readingType);
+    });
+    sensor1.onerror = t.step_func_done(event => {
+      assert_unreached(event.error.name + ":" + event.error.message);
+    });
+  }, "Test that sensor reading is correct.");
+
+  async_test(t => {
+    let sensor = new sensorType();
+    sensor.start();
+    let cachedReading1;
+    sensor.onactivate = event => {
```

Comment:
Change onactivate to onchange.
The first to get reading value, the second to get the reading value 1 second later.